File: Sample/BasicSample/sample_cpp/File/File_StaticFile.cpp

#include <Altseed.h>

// StaticFile を用いて、ファイルからテキストを読み込むサンプルです。
void File_StaticFile()
{
	// Altseedを初期化する。
	asd::Engine::Initialize(asd::ToAString("StaticFile").c_str(), 640, 480, asd::EngineOption());

	// フォントを生成する。
	auto font = asd::Engine::GetGraphics()->CreateFont(asd::ToAString("Data/Font/Font1.aff").c_str());

	// オブジェクトを生成する。
	auto obj = std::make_shared<asd::TextObject2D>();

	// 描画に使うフォントを設定する
	obj->SetFont(font);

	// 描画位置を指定する
	obj->SetPosition(asd::Vector2DF(100, 100));

	// ファイルオブジェクト作成
	auto staticFile = asd::Engine::GetFile()->CreateStaticFile(asd::ToAString("Data/Text/HelloWorld.txt").c_str());

	// 描画したいテキストをファイルから読み取る
	char text[32] = { 0 };
	memcpy(text, staticFile->GetBuffer().data(), staticFile->GetBuffer().size());

	// 描画する文字列の指定
	obj->SetText(asd::ToAString(text).c_str());

	// オブジェクトのインスタンスをエンジンへ追加する。
	asd::Engine::AddObject2D(obj);

	// Altseedのウインドウが閉じられていないか確認する。
	while (asd::Engine::DoEvents())
	{
		// Altseedを更新する。
		asd::Engine::Update();
	}

	// Altseedを終了する。
	asd::Engine::Terminate();
}

File: Sample/BasicSample/sample_cs/File/File_StaticFile.cs
using System.Text;

namespace sample_cs
{
	class File_StaticFile : ISample
    {
        public string Description
        {
            get { return ""; }
        }

        // StaticFile を用いて、ファイルからテキストを読み込むサンプルです。
        public void Run()
		{
			// Altseedを初期化する。
			asd.Engine.Initialize("File_StaticFile", 640, 480, new asd.EngineOption());

			// フォントを生成する。
			var font = asd.Engine.Graphics.CreateFont("Data/Font/Font1.aff");

			// オブジェクトを生成する。
			var obj = new asd.TextObject2D();

			// 描画に使うフォントを設定する
			obj.Font = font;

			// 描画位置を指定する
			obj.Position = new asd.Vector2DF(100, 100);

			// ファイルオブジェクト作成
			var staticFile = asd.Engine.File.CreateStaticFile("Data/Text/HelloWorld.txt");

			// 描画したいテキストのバイト列をファイルから読み取る
			byte[] textBytes = staticFile.Buffer;

			// UTF-8でバイト列をデコードする
			char[] text = Encoding.UTF8.GetChars(textBytes);

			// 描画する文字列の指定
			obj.Text = string.Concat(text);

			// オブジェクトのインスタンスをエンジンへ追加する。
			asd.Engine.AddObject2D(obj);

			// Altseedのウインドウが閉じられていないか確認する。
			while (asd.Engine.DoEvents())
			{
				// Altseedを更新する。
				asd.Engine.Update();
				Recorder.TakeScreenShot("File_StaticFile", 30);
			}

			// Altseedを終了する。
			asd.Engine.Terminate();
		}
	}
}
